main.nf

```diff
@@ -3,37 +3,46 @@
 // initialise required channels 
 DATASET_IDS = Channel.create()
 NUM_CLUST = Channel.create()
-DATA_TYPE = Channel.create()
-NORM_METHOD = Channel.create()
 BARCODE_COLUMN = Channel.create()
 CELL_LABEL_COLUMN = Channel.create()
 
-// parse txt file with dataset accessions into a queue channel 
-Channel
-    .fromPath(params.data_import.training_dataset_ids)
-    .splitCsv(header:false, sep:",")
-    .separate(DATASET_IDS, DATA_TYPE, NORM_METHOD, NUM_CLUST, BARCODE_COLUMN, CELL_LABEL_COLUMN)
+// extract matrix types required by the tools, conditioned on them being 'on' 
+tool_switch = ["True":0, "False":1]
+garnett_matrix_type = [params.garnett.matrix_type, null][tool_switch[params.garnett.run]]
```

Comment:
At some point it would be nice to re-do this logic so the methods didn't have to be hard-coded. So we could just assemble an active_methods array by parsing the params element and loop over tha, but this works right now.

Reply:
I see - we don't actually need these variables, can create a channel directly using a loop.